### PR TITLE
Eahw 2824/use GitHub packages

### PR DESCRIPTION
**Pull request by JWeston-HO**
This Pr Introduces:

-     Ability to pull in github packages
-     Updates person-restapi to use the most recent version of kafka-commons

### CODE REVIEW

File: person_settings.xml [hunk 1]
```diff
@@ -1,28 +1,32 @@
...
     <profiles>
         <profile>
-            <id>artifactory</id>
+            <id>github</id>
             <repositories>
                 <repository>
-                    <id>callisto-jparest</id>
-                    <name>Callisto Jparest Repo</name>
-                    <url>https://artifactory.digital.homeoffice.gov.uk/artifactory/callisto-jparest</url>
+                    <id>github-packages</id>
```
Comment (taomoh) [inline person_settings.xml:19]: I am surprised the 2 repositories having the same id didn't cause any problem

File: src/test/resources/application.properties [hunk 2]
```diff
@@ -1,4 +1,4 @@
-projectVersion=1.0.0
+schemaVersion=1.0.0
```
Comment (taomoh) [inline src/test/resources/application.properties:1]: Looks like src properties file still refers to projectVersion

File: pom.xml [hunk 3]
```diff
@@ -107,13 +108,24 @@
...
             <groupId>org.springframework.kafka</groupId>
             <artifactId>spring-kafka-test</artifactId>
-            <version>3.0.2</version>
+            <version>${spring.kafka.version}</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>com.fasterxml.jackson.datatype</groupId>
+            <artifactId>jackson-datatype-hibernate5-jakarta</artifactId>
+            <version>2.15.2</version>
```
Comment (taomoh) [inline pom.xml:123]: I believe scope test would suffice here

File: src/test/resources/application.properties (hunk 2 above)
Comment (JWeston-HO) [reply src/test/resources/application.properties:1]: updated to schema version

File: pom.xml (hunk 3 above)
Comment (JWeston-HO) [reply pom.xml:123]: change scope to test

File: person_settings.xml (hunk 1 above)
Comment (JWeston-HO) [reply person_settings.xml:19]: It is the ID of the server. See Maven docs for server
"id: This is the ID of the server (not of the user to login as) that matches the id element of the repository/mirror that Maven tries to connect to."

https://maven.apache.org/settings.html

Comment (taomoh) [reply person_settings.xml:19]: The excerpt you referred to is for the server element definition.
The id under repository element is the ID for the repository:
https://maven.apache.org/ref/3.9.2/maven-settings/settings.html#class_repository

It's just unexpected that the id element isn't fulfilling its implied function of identification

As it doesn't seem to break things, I am happy to resolve this comment

**Review by taomoh — APPROVED** [review]

**Review by sewerynmi — APPROVED** [review]

**Pull request merged**